pages/idt.adoc

```diff
@@ -0,0 +1,142 @@
+---
+title: Interrupt descriptor table
+tags: x86, x64
+category: Interrupts
+description: IDT specification
+---
+:source-language: c
+
+== Interrupt Descriptor Table
+
+// TODO: articles "GDT" and "Interrupts" are not written yet
+
+Interrupt Descriptor Table is a binary data structure specific to IA-32 (xref:x86[x86]) and x86_64 architectures.
+It is used by the CPU to lookup an interrupt service routines (ISR) when an xref:interrupts[interrupt] is issued.
+It is protected mode and long mode counterpart to real mode interrupt vector table (IVT).
+
+IMPORTANT: IDT requires a working xref:gdt[GDT]
+
+=== IDTR
+IDTR is a register containing the IDT's address and limit.
+
+[cols="1,3"]
+|===
+| Size
+| One less than the size of the IDT in bytes
+
+| Offset
+| The linear address of the IDT
+|===
+
+==== Layout
+
+[cols="1,1,1"]
+|===
+| Architecture
+| Size
+| Offset
+
+| IA-32
+| 16 (0 -- 15)
+| 31 (16 -- 48)
+
+| x86_64
+| 16 (0 -- 15)
+| 63 (16 -- 79)
+|===
+
+==== Things to know
+* IDT can contain up to 256 (0..255) ISR vectors. Entries above this limit will be ignored.
+* In contrast to GDT, the first entry of the IDT is valid and used.
+
+=== IDT entry
+
+
+==== IA-32
+On 32-bit processors, IDT entries are 8-bytes long, therefore to access a particular entry the raw index must be multiplied by 8 and added to the IDTR Offset.
```

Comment:
"offset" definitely lower case
comma between "entry" and "the"

Reply:
Offset is made here upper case on purpose, as it's a field of the IDTR descriptor.